pkg/api/persistentvolumeclaim/util.go

```diff
@@ -74,6 +74,16 @@ func EnforceDataSourceBackwardsCompatibility(pvcSpec, oldPVCSpec *core.Persisten
 	}
 }
 
+func DropDisabledFieldsFromStatus(pvc, oldPVC *core.PersistentVolumeClaim) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.ExpandPersistentVolumes) && oldPVC.Status.Conditions == nil {
+		pvc.Status.Conditions = nil
+	}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) && !allocatedResourcesInUse(oldPVC) {
+		pvc.Status.AllocatedResources = nil
```

Comment:
if feature is disabled, should we always set AllocatedResources to nil?

Reply:
The convention I think is - if feature is disabled and field is not in-use then and only then we should set the field to nil. This way, if feature gets enabled again - the old data will be back and not get lost.

Reply:
but this a new field, so should not have data yet?

Reply:
I think guidance I got from api reviewers is - a user may enable the feature gate and disable it and then can enable it back. It can be done during beta phase too. We want to prevent any data loss in those cases.